design/fips-compliance.md

```diff
@@ -0,0 +1,259 @@
+
+From the table above, I think that we should release only the FIPS compliant
+version. The advantages that customers will get by using the `golang` image
+as the base image is not worth the maintenance burden that releasing both
+versions will bring to the development team.
+
+Note: If the performance of the `goboring/golang` version will be much worse than that
+      of the `golang` version (degradation of more than 5%) then we will release both versions.
+      More info on the performance tests that will be done in this effort can be found in the
+      [Performance section](#performance)
+
+### Red Hat image
+
+We will also release a FIPS compliant Red Hat image of the `conjur-authn-k8s
+-client` to our Red Hat registry. Before we do that, we should create a 
+`conjur-authn-k8s-client` FIPS project in the RH container registry.
+
+TBD - how do we create the FIPS project in the RH container registry?
+
+### Performance
+
+Same SLA of performance should be kept. As of today, we don't have any
+performance tests and an SLA (that I know of). We should decide on one and
+verify that the change to GoBoring doesn't impact the performance.
+
+We should write a performance test and run it twice - once on the `master` 
+image and once on the new FIPS-compliance image - and verify that the
+ performance is not affected by the change. We can live with a degradation of less
+ than 5% between the versions.
+
+This performance test should be (in high level):
+  - Deploy a pod with the `conjur-authn-k8s-client` image and an application container
+    that retrieves a secret from Conjur using the access token retrieved by the authenticator client.
+  - Test how long the procedure above took
+  - Run this for 1000 times and get the average time
+
+## Test plan
+
+Before we dig into which tests should run, we should decide where, and how, they
+should run. Currently, the `conjur-authn-k8s-client` have UTs in its project
+and a vanilla test that runs in `kubernetes-conjur-demo`. The `kubernetes-conjur-demo`
+is triggered only when we push code to that repository. 
+
+Even while putting aside the fact that we have only a vanilla test, our current
+ test flow is still not optimal. 
+ Let's propose some options to improve our flow, and decide on the best one.
+
+### Trigger `kubernetes-conjur-demo` to run after every build of `conjur-authn-k8s-client`
+
+We can find a way to run `kubernetes-conjur-demo` using the currently built
+ authenticator-client, utilizing the `LOCAL_AUTHENTICATOR` environment variable.
+ If this is possible, we can trigger a `kubernetes-conjur-demo` build as part
```

Comment:
From meeting:
Geri - demo runs daily and on changes but only pulls from `latest` tag of project